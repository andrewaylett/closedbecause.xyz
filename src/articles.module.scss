.page {
    padding: 1ex;
    font-variant-numeric: oldstyle-nums;
    display: grid;
    grid-template-columns: 1fr min(65ch, 100%) 1fr;
}

.page > * {
    grid-column: 2;
}

.revision {
    font-size: smaller;
}

.revision::before {
    content: 'Â (';
}

.revision::after {
    content: ')';
}
