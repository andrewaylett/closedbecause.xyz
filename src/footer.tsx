import React from 'react'
import { getYear } from 'date-fns'
import * as t from 'io-ts'
import { fc_props } from './types'

const FooterProps = t.partial(
  {
    author: t.string,
    copyright: t.string,
  },
  'FooterProps'
)
type FooterProps = t.TypeOf<typeof FooterProps>

const Footer: React.FunctionComponent<FooterProps> = fc_props(
  ({ author, copyright }) => (
    <footer style={{ paddingTop: '1em' }}>
      Copyright Â© {author || 'Andrew Aylett'}, {copyright || getYear(Date.now())}
    </footer>
  ),
  FooterProps
)

export default Footer
