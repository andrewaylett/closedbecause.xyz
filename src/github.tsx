export const github_url = (page_name: string): string =>
  `https://github.com/andrewaylett/aylett.co.uk/commits/main/src/pages${page_name}.mdx`
