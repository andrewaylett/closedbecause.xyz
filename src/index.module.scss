
.container {
    min-height: 100vh;
    padding: 0 0.5rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    main {
        padding: 5rem 0;
        flex: 1;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
    }

    footer {
        width: 100%;
        height: 100px;
        border-top: 1px solid #eaeaea;
        display: flex;
        justify-content: center;
        align-items: center;

        img {
            margin-left: 0.5rem;
        }

        a {
            display: flex;
            justify-content: center;
            align-items: center;
            text-decoration: none;
            color: inherit;
        }
    }
}

%centre {
    text-align: center;
}

.title {
    @extend %centre;
    margin: 0;
    line-height: 1.15;
    font-size: 4rem;
}

.description {
    @extend %centre;
    line-height: 1.5;
    font-size: 1.5rem;
}

.logo {
    height: 1em;
}
