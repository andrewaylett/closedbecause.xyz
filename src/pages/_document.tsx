import Document, { DocumentContext, DocumentInitialProps, Head, Html, Main, NextScript } from 'next/document'
import React from 'react'

class MyDocument extends Document {
  static async getInitialProps(ctx: DocumentContext): Promise<DocumentInitialProps> {
    const originalRenderPage = ctx.renderPage

    ctx.renderPage = () => {
      return originalRenderPage({
        enhanceApp: (App) => App,
        enhanceComponent: (Component) => Component,
      })
    }

    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render(): React.ReactElement {
    // noinspection HtmlRequiredTitleElement
    return (
      <Html lang="en">
        <Head />
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

// noinspection JSUnusedGlobalSymbols
export default MyDocument
