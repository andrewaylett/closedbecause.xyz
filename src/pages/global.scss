html, body {
    padding: 0;
    margin: 0;
    font-family: Calibri, Roboto, -apple-system, BlinkMacSystemFont, Segoe UI, Oxygen,
    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

* {
    box-sizing: border-box;
}

div.mdx {
    padding: 1ex;
    font-variant-numeric: oldstyle-nums;
    display: grid;
    grid-template-columns: 1fr min(65ch, 100%) 1fr;
}
div.mdx > * {
    grid-column: 2;
}
.full-bleed {
    width: 100%;
    grid-column: 1 / 4;
}

code {
    font-variant-numeric: tabular-nums;
    font-family: "JetBrains Mono", Lucida Console, Liberation Mono, DejaVu Sans Mono, Bitstream Vera Sans Mono, monospace
}

header h1 {
    font-size: 3em;
}

.revisions {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    column-gap: 1ex;
    a {
        color: inherit;
    }
}

nav {
    padding-top: 1ex;
}

.meta {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap-reverse;
    justify-content: space-between;
    margin-top: 1ex;
}
