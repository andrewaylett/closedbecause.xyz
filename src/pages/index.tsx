import Head from 'next/head'
import styles from '../index.module.scss'
import Link from 'next/link'
import React from 'react'
import style from '../articles.module.scss'
import { ArticlesProps, fc_props } from '../types'
import { InferGetStaticPropsType } from 'next'
import { getStaticProps } from '../ssr/articles'

export { getStaticProps } from '../ssr/articles'

// noinspection HtmlUnknownTarget
export const Home: React.VoidFunctionComponent<InferGetStaticPropsType<typeof getStaticProps>> = fc_props(({ pages }) => (
  <div className={styles.container}>
    <Head>
      <title>aylett.co.uk</title>
      <link rel="icon" href="/favicon.ico" />
    </Head>

    <main>
      <h1 className={styles.title}>Closed Because...</h1>

      <p className={styles.description}>Insert &apos;90s &ldquo;site under construction&rdquo; gif here.</p>
      <div className={styles.description}>
        {
          pages.map(({ name, metadata }) => (
          <p key={name}>
            <Link href={`/${name}`}>{metadata.title}</Link>
            {metadata.author ? ` - ${metadata.author}` : ''}
            {metadata.revision ? <span className={style.revision}>{`v${metadata.revision}`}</span> : ''}
            {metadata.abstract ? `: ${metadata.abstract}` : ''}
          </p>
        ))
        }
      </div>
    </main>

    <footer>
      <a href="/_logs" target="_blank" rel="noopener noreferrer">
        Powered by <img src="/vercel.svg" alt="Vercel Logo" className={styles.logo} />
      </a>
    </footer>
  </div>
), ArticlesProps)

export default Home
