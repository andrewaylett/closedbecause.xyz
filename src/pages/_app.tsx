import App from 'next/app'
import 'normalize.css'
import './global.scss'
import React from 'react'
import { AppProps } from 'next/dist/pages/_app'
import { MDXProvider, MDXProviderComponentsProp } from '@mdx-js/react'
import Link from 'next/link'
import Footer from '../footer'
import Head from 'next/head'
import { github_url } from '../github'
import { fc_props, PageMetadata } from '../types'
import { PathReporter } from 'io-ts/PathReporter'
import * as t from 'io-ts'
import { ReactChild } from 'prop-types-ts'

const OptionalProps = t.partial({ text: t.string }, 'OptionalProps')

const Optional = fc_props(({ text, children }) => <>{text ? <span>{children}</span> : null}</>, OptionalProps)

const RevisionsProps = t.intersection([PageMetadata, t.interface({ url: t.string })], 'RevisionsProps')

const Revisions = fc_props(
  ({ revision, revised, expires, url }) => (
    <>
      {revision || revised || expires ? (
        <div className="revisions">
          <Optional text={revision}>Version:&nbsp;{revision}</Optional>
          <Optional text={revised}>
            <a href={github_url(url)}>Last Revised:&nbsp;{revised}</a>
          </Optional>
          <Optional text={expires}>Expires:&nbsp;{expires}</Optional>
        </div>
      ) : (
        ''
      )}
    </>
  ),
  RevisionsProps
)

const WrapperProps = t.interface(
  {
    metadata: PageMetadata,
    children: ReactChild,
  },
  'WrapperProps'
)
type WrapperProps = t.TypeOf<typeof WrapperProps>

const components: (url: string) => MDXProviderComponentsProp = (url) => ({
  wrapper: fc_props((props: WrapperProps) => {
    const { children, metadata: hopefullyMetadata } = props
    const decoded = PageMetadata.decode(hopefullyMetadata)
    if (decoded._tag !== 'Right') {
      throw PathReporter.report(decoded)
    }
    const metadata = decoded.right
    return (
      <div className="mdx">
        <Head>
          <title>{metadata.title} - aylett.co.uk</title>
        </Head>
        <nav>
          <Link href="/">Home</Link>
        </nav>
        <header>
          <h1>{metadata.title}</h1>
          {metadata.abstract ? metadata.abstract : ''}
          <div className="meta">
            {metadata.author ? <div className="author">Author: {metadata.author}</div> : ''}
            <Revisions url={url} {...metadata} />
          </div>
        </header>
        <main>{children}</main>
        <Footer author={metadata.author} copyright={metadata.copyright || metadata.revised.split('/')[0]} />
      </div>
    )
  }, WrapperProps),
})

const WrappedApp: React.VoidFunctionComponent<AppProps> = (props, context) => (
  <MDXProvider components={components(props.router.asPath)}>
    <App {...props} context={context} />
  </MDXProvider>
)

// noinspection JSUnusedGlobalSymbols
export default WrappedApp
