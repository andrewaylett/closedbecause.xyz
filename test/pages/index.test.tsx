import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../src/pages'
import { describe, it, expect } from '@jest/globals'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Home pages={[]} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
