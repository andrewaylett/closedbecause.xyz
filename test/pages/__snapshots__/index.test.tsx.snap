// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Closed Because...
      </h1>
      <p
        class="description"
      >
        Insert '90s “site under construction” gif here.
      </p>
      <div
        class="description"
      />
    </main>
    <footer>
      <a
        href="/_logs"
        rel="noopener noreferrer"
        target="_blank"
      >
        Powered by 
        <img
          alt="Vercel Logo"
          class="logo"
          src="/vercel.svg"
        />
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
